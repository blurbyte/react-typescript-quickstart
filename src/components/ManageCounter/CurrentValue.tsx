import styled from '@styles/styledComponents';
import { Color } from '@styles/colors';

const CurrentValue = styled.p`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  line-height: 1;
  height: 3.6rem;
  font-size: 2.4rem;
  font-weight: 700;
  color: ${Color.White};
  background: ${Color.Blue};
  padding: 0 1.4rem;
  border-radius: 6px;
  min-width: 8rem;
  margin: 1rem 0;
`;

export default CurrentValue;
