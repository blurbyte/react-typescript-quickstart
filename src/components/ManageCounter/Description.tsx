import styled from '@styles/styledComponents';
import MainParagraph from '../MainParagraph';

const Description = styled(MainParagraph)`
  flex: 1 0;
`;

export default Description;
