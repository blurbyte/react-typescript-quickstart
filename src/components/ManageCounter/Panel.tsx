import styled from '@styles/styledComponents';
import { Colors } from '@styles/colors';

const Panel = styled.div`
  width: 40rem;
  border-top: 1px solid ${Colors.gray};
  margin-top: 3rem;
`;

export default Panel;
