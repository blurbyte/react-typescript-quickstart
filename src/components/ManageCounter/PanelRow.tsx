import styled from 'styled-components';

const PanelRow = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export default PanelRow;
