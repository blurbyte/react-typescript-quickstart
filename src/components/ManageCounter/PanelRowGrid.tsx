import styled from '@styles/styledComponents';

const PanelRowGrid = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 1rem;
`;

export default PanelRowGrid;
