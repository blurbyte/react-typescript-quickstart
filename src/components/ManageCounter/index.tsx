import * as React from 'react';

import MainParagraph from '../MainParagraph';
import Headline from '../Headline';
import Counter from '../Counter';
import Button from '../Button';
import Section from '../Section';
import CurrentValue from './CurrentValue';
import Panel from './Panel';
import PanelRow from './PanelRow';
import Description from './Description';

class ManageCounter extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <Section>
        <Headline>Manage Counter</Headline>
        <MainParagraph>
          A strigtforward counter example built using popular <em>render props</em> pattern.
        </MainParagraph>
        <Counter initialCount={10} step={2}>
          {({ count, initialCount, increase, decrease }) => (
            <Panel>
              <PanelRow>
                <Description>Current Counter Value</Description>
                <CurrentValue>{count}</CurrentValue>
              </PanelRow>
              <PanelRow>
                <Button onClick={increase}>increase</Button>
                <Button disabled={count <= initialCount} onClick={decrease}>
                  decrease
                </Button>
              </PanelRow>
            </Panel>
          )}
        </Counter>
      </Section>
    );
  }
}

export default ManageCounter;
