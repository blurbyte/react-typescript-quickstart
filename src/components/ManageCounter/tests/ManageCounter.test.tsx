import * as React from 'react';
import { shallow } from 'enzyme';

import ManageCounter from '../index';

describe('ManageCounter', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ManageCounter />);
    expect(wrapper).toMatchSnapshot();
  });
});
