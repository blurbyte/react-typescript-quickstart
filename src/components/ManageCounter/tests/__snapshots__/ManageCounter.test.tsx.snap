// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageCounter renders correctly 1`] = `
<styled.section>
  <styled.h1>
    Manage Counter
  </styled.h1>
  <styled.p>
    A strigtforward counter example built using popular 
    <em>
      render props
    </em>
     pattern.
  </styled.p>
  <Counter
    initialCount={10}
    step={2}
  />
</styled.section>
`;
